Backendless/Persistence/GeoJSONParser.cs

```diff
@@ -0,0 +1,158 @@
+    {
+    }
+
+    public GeoJSONParser( ReferenceSystemEnum srs, String geomClassName )
+    {
+      this.srs = srs;
+      if ( geomClassName != null )
+      {
+        try
+        {
+          Assembly asm = Assembly.GetExecutingAssembly();
+          T unchekedClazz = ( T )asm.CreateInstance( geomClassName );
+          this.geomClass = unchekedClazz;
+        }
+        catch
+        {
+          throw new ArgumentException( $"'geomClassName' contains uknown class '{geomClassName}'." );
+        }
+      }
+      else
+        this.geomClass = null;
+    }
+
+    public Geometry Read( String geoJSON )
+    {
+      if ( geoJSON == null )
+        return null;
+
+      Dictionary<string, object> geoJSONMap;
+      try
+      {
+        geoJSONMap = (new Json()).Deserialize( geoJSON );
+      }
+      catch( System.Exception ex )
+      {
+        throw new GeoJSONParserException( ex );
+      }
+      return Read( geoJSONMap );
+    }
+
+    public Geometry Read( Dictionary<string, object> geoJSON )
+    {
+      string type = ( string )geoJSON[ "type" ];
+      Object coordinatesObj = geoJSON["coordinates"];
+
+      Object[] coordinates = null;
+
+      if ( coordinatesObj is List<Object> )
+        coordinates = ( ( List<Object> )coordinatesObj ).ToArray();
+      else if ( coordinatesObj != null )
+        coordinates = ( ( List<Double> )coordinatesObj ).Select( d => ( Object )d ).ToArray();
+       
+      if ( type == null || coordinates == null )
+        throw new GeoJSONParserException( "Both 'type' and 'coordinates' should be present in GeoJSON object." );
+      if ( this.geomClass == null || this.geomClass.GetType() == typeof( Geometry ) || typeof( Point ) == this.geomClass.GetType()
+          || typeof( LineString ) == this.geomClass.GetType() || typeof( Polygon ) == this.geomClass.GetType() )
+      {
+        switch ( type )
+        {
+          case Point.GEOJSON_TYPE:
+            return ConstructPointFromCoordinates( coordinates );
+          case LineString.GEOJSON_TYPE:
+            return ConstructLineStringFromCoordinates( coordinates );
+          case Polygon.GEOJSON_TYPE:
+            return ConstructPolygonFromCoordinates( coordinates );
+        }
+      }
+      else
+        throw new GeoJSONParserException( $"Unknown geometry class: '{this.geomClass}" );
+
+      throw new GeoJSONParserException( $"Unknown geometry type: '{type}'" );
+    }
+
+    private Point ConstructPointFromCoordinates( Object[] coordinatePair )
+    {
+      return new Point( this.srs ).SetX( ( double )coordinatePair[0] ).SetY( ( double )coordinatePair[1] );
+    }
+
+    private LineString ConstructLineStringFromCoordinates( Object[] arrayOfCoordinatePairs)
+    {
+      List<Point> points = new List<Point>();
+
+      Object[] coordinatePairNumbers;
+      foreach(Object coordinatePairObj in arrayOfCoordinatePairs)
+      {
+        if ( coordinatePairObj is List<Object> )
```

Comment:
can you describe the logic? in what case it is `Lis<Object>` and when it is not?